overlord/devicestate/devicemgr.go

```diff
@@ -148,31 +149,83 @@ func (m *DeviceManager) ensureOperational() error {
 		return nil
 	}
 
+	// conditions to trigger device registration
+	//
+	// * have a model assertion with a gadget (core and
+	//   device-like classic) in which case we need also to wait
+	//   for the gadget to have been installed though
+	// * classic with a model assertion with a non-default store specified
+	// * lazy classic case (might have a model with no gadget nor store
+	//   or no model): we wait to have some snaps installed or be
+	//   in the process to install some
+
+	var seeded bool
+	err = m.state.Get("seeded", &seeded)
+	if err != nil && err != state.ErrNoState {
+		return err
+	}
+
 	if device.Brand == "" || device.Model == "" {
-		// cannot proceed until seeding has loaded the model
-		// assertion and set the brand and model, that is
-		// optional on classic
-		// TODO: later we can check if
-		// "seeded" was set and we still don't have a brand/model
-		// and use a fallback assertion
-		return nil
+		if !release.OnClassic || !seeded {
+			return nil
+		}
+		// we are on classic and seeded but there is no model:
+		// use a dummy model!
+		// TODO: have a real matching model assertion?  use a
+		// different brand (requires store support!)?
+		device.Brand = "canonical"
+		device.Model = "generic-classic"
+		err := auth.SetDevice(m.state, device)
```

Comment:
(nitpick^2) this could be written in a single line: `if err := auth.SetDevice(..); err != nil {`